### PR TITLE
various: Updates for proper syncing via grpc for version 5.0.0

### PR DESCRIPTION
**Pull request by alexlyp**
This PR covers various changes that allow dcrwallet to be synced via requests over grpc in normal mode.  Here are some of those changes:  

- New requests `FetchMissingCFilters` and `RescanPoint`
- New request field of `starting_block_hash` in `DiscoverAddressRequest`
- Removed `chainClient` from `makeTicketSummary`.

`FetchMissingCFilters` is now required due to both spv and normal mode using commited filters for proper syncing.  

`RescanPoint` is a new request that provides the last place the wallet had been rescanned.  This allows for more efficient rescanning and discovering addresses requests.

`DiscoverAddressRequest` now makes use of the `RescanPoint` response

### CODE REVIEW

File: rpc/documentation/api.md [hunk 1]
```diff
@@ -304,6 +306,8 @@ for querying for known used addresses).  Account discovery requires the wallet
 to be unlocked in order to derive account hardened extended pubkeys, and thus
 the private passphrase must be passed as a parameter when performing this
 action.  Account discovery is typically only required when reseeding a wallet.
+The address discover will begin at the provided starting block hash.  If none
```
Comment (jrick) [inline rpc/documentation/api.md]: Address discovery begins at...

File: rpc/documentation/api.md [hunk 2]
```diff
@@ -304,6 +306,8 @@ for querying for known used addresses).  Account discovery requires the wallet
...
+is provided, then the current network's GenesisHash will be used.
```
Comment (jrick) [inline rpc/documentation/api.md]: "genesis hash". Use plain english in documentation instead of identifiers from the code unless specifically referring specifically to that identifier.

File: rpc/documentation/api.md [hunk 3]
```diff
@@ -313,14 +317,18 @@ action.  Account discovery is typically only required when reseeding a wallet.
 - `bytes private_passphrase`: The private passphrase to unlock the wallet when
   account discovery is enabled.
 
+- `bytes starting_block_hash`: The hash at which Discover Addresses should 
+  begin.
```
Comment (jrick) [inline rpc/documentation/api.md]: Reads better as "The hash of the first block checked for address and account usage."

Please also document that this must be a main chain block.

File: rpc/documentation/api.md [hunk 4]
```diff
@@ -376,6 +384,45 @@ should begin at.
 - `FailedPrecondition`: The wallet or consensus RPC server has not been opened.
 
 **Stability:** Unstable
+___
+
+#### `FetchMissingCFilters`
+
+The `FetchMissingCFilters` method fetches any missing committed filters from
+the main chain.
```
Comment (jrick) [inline rpc/documentation/api.md]: "...committed filters for blocks in the wallet's main chain".

File: rpc/documentation/api.md [hunk 5]
```diff
@@ -376,6 +384,45 @@ should begin at.
...
+**Expected errors:**
+
+- `FailedPrecondition`: The wallet or consensus RPC server has not been opened.
+
+**Stability:** Unstable
+___
+
+#### `RescanPoint`
+
+The `RescanPoint` method returns the current RescanPoint hash which allows other
```
Comment (jrick) [inline rpc/documentation/api.md]: self-referential documentation isn't particularly useful.  This should describe what the rescan point is, not that it returns it.

For example (taken from the Wallet.RescanPoint godoc comment):

>  RescanPoint returns the block hash at which a rescan should begin (inclusive), or nil when no rescan is necessary.

Replace nil with null since this is protobuf not go.

File: rpc/documentation/api.md [hunk 6]
```diff
@@ -376,6 +384,45 @@ should begin at.
...
+methods to know where the wallet has last been rescanned through.  This allows
+clients to easily discern where startup rescan should be started from or
+whether DiscoverAddresses is required to run again.
```
Comment (jrick) [inline rpc/documentation/api.md]: This second sentence could read "A non-null rescan point indicates that blocks currently in the main chain must be checked for address usage and relevant transactions."

File: rpc/documentation/api.md [hunk 7]
```diff
@@ -376,6 +384,45 @@ should begin at.
...
+
+**Request:** `RescanPointRequest`
+
+**Response:** `RescanPointResponse`
+
+- `bytes rescan_point_hash`: The current hash of the RescanPoint.
```
Comment (jrick) [inline rpc/documentation/api.md]: The field can just be `rescan_point`.  Don't refer to the Go identifier here either.

File: rpc/documentation/api.md [hunk 8]
```diff
@@ -376,6 +384,45 @@ should begin at.
...
+
+**Expected Errors:**
+
+- `FailedPrecondition`: The wallet or consensus RPC server has not been opened.
+
+**Stability:** Unstable
+___
+___
+___
```
Comment (jrick) [inline rpc/documentation/api.md]: We don't need these separators.  there was previously no separator between the last documented method and the header for the next service.

File: rpc/rpcserver/log.go [hunk 9]
```diff
@@ -23,6 +23,8 @@ import (
 	"google.golang.org/grpc/grpclog"
 )
 
+var log = slog.Disabled
```
Comment (jrick) [inline rpc/rpcserver/log.go]: why is this added?  the service implementations don't log anything at all currently.  this doesn't even add a way to modify the logger to be anything but the disabled logger.

File: rpc/rpcserver/server.go [hunk 10]
```diff
@@ -57,10 +57,10 @@ import (
 
 // Public API version constants
 const (
-	semverString = "5.0.0"
+	semverString = "5.0.1"
```
Comment (jrick) [inline rpc/rpcserver/server.go]: This change is a minor API bump, not patch.  The bump must also be recorded in the api.md document.

File: rpc/rpcserver/server.go [hunk 11]
```diff
@@ -2285,16 +2294,56 @@ func (s *loaderServer) DiscoverAddresses(ctx context.Context, req *pb.DiscoverAd
...
+		}
+	}
+	err = wallet.DiscoverActiveAddresses(ctx, n, startHash, req.DiscoverAccounts)
 	if err != nil {
 		return nil, translateError(err)
 	}
 
 	return &pb.DiscoverAddressesResponse{}, nil
 }
+func (s *loaderServer) FetchMissingCFilters(ctx context.Context, req *pb.FetchMissingCFiltersRequest) (
```
Comment (jrick) [inline rpc/rpcserver/server.go]: need an empty line between this function and the previous

File: rpc/rpcserver/server.go [hunk 12]
```diff
@@ -2285,16 +2294,56 @@ func (s *loaderServer) DiscoverAddresses(ctx context.Context, req *pb.DiscoverAd
...
+	n := chain.BackendFromRPCClient(s.rpcClient.Client)
+	// Fetch any missing main chain compact filters.
+	err := wallet.FetchMissingCFilters(ctx, n)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.FetchMissingCFiltersResponse{}, nil
+}
+func (s *loaderServer) RescanPoint(ctx context.Context, req *pb.RescanPointRequest) (*pb.RescanPointResponse, error) {
```
Comment (jrick) [inline rpc/rpcserver/server.go]: newline

File: rpc/rpcserver/server.go [hunk 13]
```diff
@@ -2285,16 +2294,56 @@ func (s *loaderServer) DiscoverAddresses(ctx context.Context, req *pb.DiscoverAd
...
+	rescanPoint, err := wallet.RescanPoint()
+	if err != nil {
+		return nil, status.Errorf(codes.FailedPrecondition, "Rescan point failed to be requested %v", err)
+	}
+	if rescanPoint != nil {
+		return &pb.RescanPointResponse{
+			RescanPointHash: rescanPoint[:],
+		}, nil
+	} else {
+		return &pb.RescanPointResponse{RescanPointHash: nil}, nil
```
Comment (jrick) [inline rpc/rpcserver/server.go]: don't return from both branches of an if-else.  use an early return for one and move the other outside of the branch.

File: rpc/rpcserver/server.go [hunk 14]
```diff
@@ -2285,16 +2294,56 @@ func (s *loaderServer) DiscoverAddresses(ctx context.Context, req *pb.DiscoverAd
...
+	}
+}
 func (s *loaderServer) SubscribeToBlockNotifications(ctx context.Context, req *pb.SubscribeToBlockNotificationsRequest) (
```
Comment (jrick) [inline rpc/rpcserver/server.go]: newline

File: rpc/rpcserver/server.go [hunk 15]
```diff
@@ -2322,7 +2371,9 @@ func (s *loaderServer) SubscribeToBlockNotifications(ctx context.Context, req *p
 	// control over how long the synchronization task runs.
 	syncer := chain.NewRPCSyncer(wallet, chainClient)
 	go syncer.Run(context.Background(), false)
-	wallet.SetNetworkBackend(chain.BackendFromRPCClient(chainClient.Client))
+
+	n := chain.BackendFromRPCClient(chainClient.Client)
+	wallet.SetNetworkBackend(n)
```
Comment (jrick) [inline rpc/rpcserver/server.go]: why this change?

File: wallet/notifications.go [hunk 16]
```diff
@@ -177,7 +176,7 @@ func makeTxSummary(dbtx walletdb.ReadTx, w *Wallet, details *udb.TxDetails) Tran
 	}
 }
 
-func makeTicketSummary(chainClient *dcrrpcclient.Client, dbtx walletdb.ReadTx, w *Wallet, details *udb.TicketDetails) *TicketSummary {
+func makeTicketSummary(dbtx walletdb.ReadTx, w *Wallet, details *udb.TicketDetails) *TicketSummary {
```
Comment (jrick) [inline wallet/notifications.go]: Removing these details obtained through dcrd RPC breaks compatibility with the previous API.

File: rpc/rpcserver/server.go (hunk 15 above)
Comment (alexlyp) [reply rpc/rpcserver/server.go]: don't recall making this change, reverting
